docs/resources/dataplatform_cluster.md

```diff
@@ -0,0 +1,293 @@
+---
+subcategory: "Data Platform"
+layout: "vkcs"
+page_title: "vkcs: vkcs_dataplatform_cluster"
+description: |-
+  Manages a dataplatform cluster
+---
+
+# vkcs_dataplatform_cluster
+
+
+
+## Example Usage
+
+# Spark
+```terraform
+data "vkcs_dataplatform_templates" "cluster-templates" {
+}
+
+resource "vkcs_dataplatform_cluster" "basic-opensearch" {
+  cluster_template_id = data.vkcs_dataplatform_templates.cluster-templates.templates[index(data.vkcs_dataplatform_templates.cluster-templates.*.product_name, "spark")].id
+  name                = "tf-basic-spark"
+  network_id          = "d2fad739-1b10-4dc8-9b2c-c246d7a7cc69"
+  subnet_id           = "3a744943-fcc1-4a85-a96b-3dc4fff71885"
+  product_name        = "spark"
+  product_version     = "3.5.1"
+
+  availability_zone = "GZ1"
+  configs = {
+    settings = [
+      {
+        alias = "sparkproxy.spark_version"
+        value = "spark-py-3.5.1:v3.5.1.2"
+      }
+    ]
+    users = [
+      {
+        user     = "user"
+        password = "somepa55word!"
+      }
+    ]
+    maintenance = {
+      start = "0 0 1 * *"
+    }
+    warehouses = [
+      {
+        name = "db_customer"
+        users = [
+          "user"
+        ]
+        connections = [
+          {
+            name = "s3_int"
+            plug = "s3-int"
+            settings = [
+              {
+                alias = "s3_bucket"
+                value = "really-cool-bucket"
```

Comment:
this bucket will be created in the cloud, so lets name it with tf-example (prefix or suffix - could you pls reffer to other examples)

another bad thing is buckets names are globally unique

Reply:
Renamed, added random suffix generation